src/Umbraco.Web/Editors/LanguageController.cs

```diff
@@ -35,59 +42,119 @@ public IEnumerable<Culture> GetAllCultures()
         [HttpGet]
         public IEnumerable<Language> GetAllLanguages()
         {
-            var allLanguages = Services.LocalizationService.GetAllLanguages();
+            var allLanguages = Services.LocalizationService.GetAllLanguages().OrderBy(x => x.Id).ToList();
+            var langs = Mapper.Map<IEnumerable<Language>>(allLanguages).ToList();
 
-            return allLanguages.Select(x => new Language
+            //if there's only one language, by default it is the default
+            if (langs.Count == 1)
+                langs[0].IsDefaultVariantLanguage = true;
+            else if (allLanguages.All(x => !x.IsDefaultVariantLanguage))
             {
-                Id = x.Id,
-                IsoCode = x.IsoCode,
-                Name = x.CultureInfo.DisplayName,
-                IsDefaultVariantLanguage = x.IsDefaultVariantLanguage,
-                Mandatory = x.Mandatory
-            });
+                //if no language has the default flag, then the defaul language is the one with the lowest id
+                langs[0].IsDefaultVariantLanguage = true;
+                langs[0].Mandatory = true;
```

Comment:
@Shazwazza here we set mandatory if we don't find any languages with `DefaultVariantLanguage` set, however above on line 50 - we only set the `DefaultVariantLanguage` and not the `Mandatory` flag. Is this correct?

Reply:
@warrenbuckley nope, that is not correct please fix :)